Tables in Galera Cluster need a Primary Key (PK)

Galera Cluster wsrep / sync can fall on its nose because of this monitoring table as it has no PK defined. This would then look something like this in the error log:

```
2021-05-01 17:05:03 19 [ERROR] InnoDB: Conflicting lock on table: `monitoring`.`monitoring` index: GEN_CLUST_INDEX that has lock 
RECORD LOCKS space id 8 page no 3 n bits 72 index GEN_CLUST_INDEX of table `monitoring`.`monitoring` trx id 627 lock_mode X locks rec but not gap
Record lock, heap no 2
Record lock, heap no 3
2021-05-01 17:05:03 19 [ERROR] InnoDB: WSREP state: 
2021-05-01 17:05:03 19 [ERROR] WSREP: Thread BF trx_id: 628 thread: 19 seqno: 247 client_state: exec client_mode: high priority transaction_mode: executing applier: 1 toi: 0 local: 0 query: UPDATE monitoring SET mytime=1619879025 WHERE host = 'node3'qd�`
2021-05-01 17:05:03 19 [ERROR] WSREP: Thread BF trx_id: 627 thread: 15 seqno: 248 client_state: exec client_mode: high priority transaction_mode: executing applier: 1 toi: 0 local: 0 query: UPDATE monitoring SET mytime=1619879025 WHERE host = 'node2'qd�`
2021-05-01 17:05:03 0x7f47684d5700  InnoDB: Assertion failure in file /home/buildbot/buildbot/build/mariadb-10.4.16/storage/innobase/lock/lock0lock.cc line 687
InnoDB: We intentionally generate a memory trap.
```

This is nicely documented on SeveralNines: https://severalnines.com/database-blog/troubleshooting-galera-cluster-issues-part-1

--- a/README.md
+++ b/README.md
@@ -4,12 +4,12 @@ A simple monitoring plugin to check if MySQL/MariaDB host can do write operation
 How does it work?
 The plugin will connect to the given MySQL server and database given by -d parameter.
 Within this database, a table "monitoring" must exist.
-Here are the SQL commands to prepare the database on a mysql server, assuming you name the database 'mymonitoring':
+Here are the SQL commands to prepare the database on a mysql server, assuming you name the database 'monitoring':
 
 ```
-CREATE DATABASE mymonitoring;
-GRANT ALL ON mymonitoring.* TO 'monitoring'@'%' IDENTIFIED BY 'secretpassword';
-CREATE TABLE mymonitoring.monitoring ( host VARCHAR(100), mytime INT(13) );
+CREATE DATABASE monitoring;
+GRANT ALL ON monitoring.* TO 'monitoring'@'%' IDENTIFIED BY 'secretpassword';
+CREATE TABLE monitoring.monitoring ( id INT(3) NOT NULL AUTO_INCREMENT, host VARCHAR(100), mytime INT(13), PRIMARY KEY (id) );
 ```
 
 Every time the plugin runs, the "mytime" column of the row matching the monitoring host will be updated with the current timestamp.

--- a/check_mysql_write.sh
+++ b/check_mysql_write.sh
@@ -7,6 +7,7 @@
 # 20190523   Adapt script for multi-master clusters (use different row per host) #
 # 20190821   Do not use password in mysql cli, use env variable                  #
 # 20200903   Add help+version, define default database                           #
+# 20210502   Add Primary Key to table (required by Galera) fixes issue #3        #
 ##################################################################################
 # Usage: ./check_mysql_write.sh -H dbhost [-P port] -u dbuser -p dbpass [-d database ]
 ##################################################################################
@@ -14,11 +15,11 @@
 # The plugin will connect to the given MySQL server and awaits a database given by -d parameter.
 # Within this database, a table "monitoring" must exist.
 # Here are the SQL commands to prepare the database on a mysql server,
-# assuming you name the database 'mymonitoring':
+# assuming you name the database 'monitoring':
 
-# CREATE DATABASE mymonitoring;
-# GRANT ALL ON mymonitoring.* TO 'monitoring'@'%' IDENTIFIED BY 'secret';
-# CREATE TABLE mymonitoring.monitoring ( host VARCHAR(100), mytime INT(13) );
+# CREATE DATABASE monitoring;
+# GRANT ALL ON monitoring.* TO 'monitoring'@'%' IDENTIFIED BY 'secret';
+# CREATE TABLE monitoring.monitoring ( id INT(3) NOT NULL AUTO_INCREMENT, host VARCHAR(100), mytime INT(13), PRIMARY KEY (id) );
 
 # Every time the plugin runs, the "mytime" column of the row matching the host will be updated with the current timestamp.
 #########################################################################
@@ -29,7 +30,7 @@ STATE_UNKNOWN=3         # define the exit code if status is Unknown
 curtime=`date +%s`
 port=3306
 database="monitoring"
-version="1.4"
+version="1.5"
 export PATH=$PATH:/usr/local/bin:/usr/bin:/bin # Set path
 
 for cmd in mysql awk grep [

